ideas/154-support-web3.js-library.md

```diff
@@ -0,0 +1,77 @@
+## Preamble
+
+    Idea: 154
+    Title: Support web3.js library
+    Status: Draft
+    Created: 2018-04-09
+    
+## Summary
+Ensure that any DApp that works in Metamask also works in Status by providing web3.js library support.
+
+## Swarm Participants
+- Lead Contributor: [@flexsurfer](https://github.com/flexsurfer)
+- Testing & Evaluation: 
+- Contributor:
+- Contributor:
+- PM: [@rachelhamlin](https://github.com/rachelhamlin)
+- UX: N/A
+
+## Product Overview
+
+A fundamental step in supporting the developer ecosystem is to ensure that any existing DApp loads and is operable in the Status browser.
+
+Users:
+1. Developers who want to develop a new HTML DApp for Ethereum from scratch
+2. Developers of existing HTML DApps that work on desktop with Metamask
+3. Users exploring DApps in Status
+
+There are a number of bugs preventing this currently. In addition to resolving any known issues, we will provide support for the web3.js library.
+
+### Product Description
+
+Developers can: 
+
+Users can load, log in to, and transact inside any DApp using the Status browser.
+
+### Requirements & Dependencies
+
+[#3174](https://github.com/status-im/status-react/issues/3174): Some Android devices can't find web3 library
+
+[#784](https://github.com/status-im/status-go/issues/784): Contract handler not called after contract deployment
+
+[#2479](https://github.com/status-im/status-react/issues/2479): web3 filters don't work with Status chat bot
+
+[#2704](https://github.com/status-im/status-react/issues/2704): Personal sign message support (CryptoKitties blocker)
+
+[#3131](https://github.com/status-im/status-react/issues/3131): Scrolling glitch in ERC dEX
+
+[ISSUE REQUIRED](https://github.com/status-im/status-react/blob/develop/resources/js/web3_init.js#L25): Sync calls are not supported in Status 
+
+ISSUE REQUIRED: [Blockchain Cuties](https://blockchaincuties.co ) does not load inside Status
+
+### Minimum Viable Product
+
+Goal date: 
+
+Description:
+- Fix known issues. 
+- Support web3.js library v.0.20.3
+
+Testing days required:
+
+## Exit Criteria
+- Test cases defined for web3.js v.0.20.3
+- Known issues closed
+
+## Success Metrics
+- All tests pass QA
+- Every Selected DApp works (loads, logs in, transacts)
```

Comment:
logs in - something not related to web3, but, maybe we can keep it, because for example you can't logs in in cryptokitties because web3

Reply:
Sure, since we are aware of the problem and anything that prevents a DApp that runs in Metamask from running in Status seems like fair game.